.github/workflows/rust.yml

```diff
@@ -27,6 +27,11 @@ jobs:
             env:
               AS_DEPENDENCY: true
               PIN_VERSIONS: true
+          - rust: 1.47
+            env:
+              AS_DEPENDENCY: true
+              PIN_VERSIONS: true
```

Comment:
Do we need to pin dependency versions for Rust 1.47?

Reply:
IMO it's better to pin them so they don't suddenly break our builds.

Reply:
My feeling is that it's better not to pin them, so we can notice when new pins are needed.

Reply:
@apoelstra I think for "is this mergeable" tests it's better to pin them but have a separate "test with newest stuff" test that is run periodically for informative purposes that tests newest dependencies as well as Rust nightly. If I remember correctly we have something like that in electrs testing against Electrum implemented usign some kind of GH Actions cron jobs.

Reply:
Ok, that sounds good to me